tailcfg/tailcfg.go

```diff
@@ -2883,3 +2883,19 @@ type EarlyNoise struct {
 // For some request types, the header may have multiple values. (e.g. OldNodeKey
 // vs NodeKey)
 const LBHeader = "Ts-Lb"
+
+// ServiceIPMappings maps service names (strings that conform to
+// [CheckServiceName]) to lists of IP addresses. This is used as the value of
+// the [NodeAttrServiceHost] capability, to inform service hosts what IP
+// addresses they need to listen on for each service that they are advertising.
+//
+// This is of the form:
+//
+//	{
+//	  "svc:samba": ["100.65.32.1", "fd7a:115c:a1e0::1234"],
+//	  "svc:web": ["100.102.42.3", "fd7a:115c:a1e0::abcd"],
+//	}
+//
+// where the IP addresses are the IPs of the VIP services that the service host
+// is expected to add to its AllowedIPs.
```

Comment:
this wording kinda implies that "the service host" (this node?) is supposed to mutate the AllowedIPs it was told about from control. I don't think that's the case? It's more like these IPs should've already been part of the AllowedIPs that were pushed down? If so, say that? And clarify the behavior if these IPs are not part of AllowedIPs? Say that's a bug and the IP should be ignored if so?

Reply:
Service host is the node that is advertising a service (so yes, this node). I guess a comment somewhere in the file (probably here) needs to define what a service host is if we're using that term.